tests/integration/test_data/private_repo_packages.yaml

```diff
@@ -8,3 +8,483 @@ private_repo_https:
+            version: null
+          - name: io/fs
+            replaces: null
+            type: go-package
+            version: null
+          - name: math
+            replaces: null
+            type: go-package
+            version: null
+          - name: math/bits
+            replaces: null
+            type: go-package
+            version: null
+          - name: os
+            replaces: null
+            type: go-package
+            version: null
+          - name: path
+            replaces: null
+            type: go-package
+            version: null
+          - name: reflect
+            replaces: null
+            type: go-package
+            version: null
+          - name: rsc.io/quote
+            replaces: null
+            type: go-package
+            version: v1.5.2
+          - name: rsc.io/sampler
+            replaces: null
+            type: go-package
+            version: v1.3.0
+          - name: runtime
+            replaces: null
+            type: go-package
+            version: null
+          - name: runtime/internal/atomic
+            replaces: null
+            type: go-package
+            version: null
+          - name: runtime/internal/math
+            replaces: null
+            type: go-package
+            version: null
+          - name: runtime/internal/sys
+            replaces: null
+            type: go-package
+            version: null
+          - name: runtime/internal/syscall
+            replaces: null
+            type: go-package
+            version: null
+          - name: sort
+            replaces: null
+            type: go-package
+            version: null
+          - name: strconv
+            replaces: null
+            type: go-package
+            version: null
+          - name: strings
+            replaces: null
+            type: go-package
+            version: null
+          - name: sync
+            replaces: null
+            type: go-package
+            version: null
+          - name: sync/atomic
+            replaces: null
+            type: go-package
+            version: null
+          - name: syscall
+            replaces: null
+            type: go-package
+            version: null
+          - name: time
+            replaces: null
+            type: go-package
+            version: null
+          - name: unicode
+            replaces: null
+            type: go-package
+            version: null
+          - name: unicode/utf8
+            replaces: null
+            type: go-package
+            version: null
+          - name: unsafe
+            replaces: null
+            type: go-package
+            version: null
+        name: github.com/cachito-testing/cachito-gomod-with-deps-private
+        type: go-package
+        version: "MAIN_VERSION"
+      - dependencies:
+          - name: github.com/cachito-testing/cachito-gomod-private
+            replaces: null
+            type: gomod
+            version: v0.0.0-20230320124445-3f12ed66d286
```

Comment:
Should we also push a new commit to the dependency repo, just to be sure that Athens won't cache it and successfully serve it even if the authentication is broken?